src/components/Form.js

```diff
@@ -171,14 +171,16 @@ class Form extends React.Component {
      * @returns {Object} - An object containing the errors for each inputID, e.g. {inputID1: error1, inputID2: error2}
      */
     validate(values) {
+        const trimmedValues = {};
+        _.each(values, (inputValue, inputID) => _.isString(inputValue) && (trimmedValues[inputID] = inputValue.trim()));
         FormActions.setErrors(this.props.formID, null);
         FormActions.setErrorFields(this.props.formID, null);
 
         // Run any validations passed as a prop
-        const validationErrors = this.props.validate(values);
+        const validationErrors = this.props.validate(trimmedValues);
```

Comment:
Will the errors for non-string values will still be validated?

Reply:
@rushatgabhane I am checking if it is a string value, it will be trimmed. So that non-string values will not be trimmed but it still is validated as before. 
We can replace `_.isString(inputValue)` into  `!_.isEmpty(inputValue)` then it will check for both string and non-string values
But I don't see any values are non-string in Form

Reply:
> But I don't see any values are non-string in Form

agree with you. but then there's a reason you're checking if the value is string before trimming.

Form is a generic component so I think we should handle non string values as well.